spec/controllers/account_subscriptions_controller_spec.rb

```diff
@@ -0,0 +1,150 @@
+        end
+
+        it "renders logged in content on the button" do
+          expect(html.at(".govuk-button").text).to eq(I18n.t("account_subscriptions.new.logged_in.action_button_text"))
+        end
+
+        it "contains create_account_subscription_path as the form action" do
+          expect(html.at("form")["action"]).to eq(create_account_subscription_path)
+        end
+      end
+
+      context "when logged out" do
+        before do
+          stub_account_api_get_sign_in_url(
+            redirect_path: new_account_subscription_path(base_path),
+            level_of_authentication: "level0",
+            auth_uri: auth_uri,
+          )
+          get :new, params: { link: base_path }
+        end
+
+        it "returns 400 if no link or topic parameter is present" do
+          get :new
+          expect(response).to have_http_status(:bad_request)
+        end
+
+        it "returns 200 if a link parameter is present" do
+          expect(response).to have_http_status(:ok)
+        end
+
+        it "renders logged out content on the title" do
+          expect(html.at("title").text).to eq("#{I18n.t('account_subscriptions.new.logged_out.title')} - GOV.UK")
+        end
+
+        it "renders logged out content on the header" do
+          expect(html.at("h1").text).to eq(I18n.t("account_subscriptions.new.logged_out.heading"))
+        end
+
+        it "renders logged out content on the description" do
+          expect(response.body).to include(I18n.t("account_subscriptions.new.logged_out.description"))
+        end
+
+        it "renders logged out content on the button" do
+          expect(html.at(".govuk-button").text).to eq(I18n.t("account_subscriptions.new.logged_out.action_button_text"))
+        end
+
+        it "adds logged out tracking values to the button" do
+          expect(html.at(".govuk-button")["data-track-action"]).to eq(I18n.t("account_subscriptions.new.logged_out.tracking"))
+        end
+
+        it "has form action pointing to authenticate endpoint" do
+          expect(html.at("form")["action"]).to eq("/sign-in")
+        end
+      end
+    end
+
+    describe "POST /email/subscriptions/account" do
+      context "when logged out" do
+        before do
+          get :create, params: { link: base_path }
+        end
+
+        it "returns 302" do
+          expect(response).to have_http_status(:redirect)
+        end
+
+        it "sends the user to new_account_subscription_path" do
+          response.should redirect_to(new_account_subscription_path({ link: base_path }))
+        end
+      end
+
+      context "when logged in" do
+        it "creates a subscriber list, links the subscription in accounts and redirects the user back to the content item" do
+          mock_logged_in_session("new-session-id")
+          stub_email_alert_api_creates_subscriber_list(
+            {
+              "title" => "Salary sacrifice",
```

Comment:
I would have preferred to use a fixture declared in this file and then re-use it.
But other tests in this repo seemed to get content items from the [govuk-content-schemas](https://github.com/alphagov/govuk-content-schemas) repo, so I did the same.

I could load these from the [detailed guide](https://github.com/alphagov/govuk-content-schemas/blob/main/examples/detailed_guide/frontend/detailed_guide.json) example too, but seemed quick and easy to just type out the expectations here. The history on the file indicates they don't change often.